components/github-api-client/src/types.rs

```diff
@@ -0,0 +1,392 @@
+    pub ssh_url: String,
+    pub clone_url: String,
+    pub default_branch: String,
+    pub master_branch: String,
+    pub organization: String,
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(default)]
+pub struct Organization {
+    pub login: String,
+    pub id: u64,
+    pub avatar_url: String,
+    pub url: String,
+    pub company: Option<String>,
+    pub description: Option<String>,
+    pub gravatar_id: Option<String>,
+    pub hooks_url: Option<String>,
+    pub html_url: Option<String>,
+    pub followers_url: Option<String>,
+    pub following_url: Option<String>,
+    pub gists_url: Option<String>,
+    pub starred_url: Option<String>,
+    pub subscriptions_url: Option<String>,
+    pub organizations_url: Option<String>,
+    pub repos_url: Option<String>,
+    pub events_url: Option<String>,
+    pub received_events_url: Option<String>,
+    pub members_url: Option<String>,
+    pub site_admin: Option<bool>,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+#[serde(default)]
+pub struct Team {
+    pub id: u64,
+    pub url: String,
+    pub name: String,
+    pub slug: String,
+    pub description: Option<String>,
+    pub privacy: String,
+    pub permission: String,
+    pub members_url: String,
+    pub repositories_url: String,
+    pub members_count: u64,
+    pub repos_count: u64,
+    pub organization: Organization,
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(default)]
+pub struct User {
+    pub login: String,
+    pub id: u64,
+    pub avatar_url: String,
+    pub gravatar_id: String,
+    pub url: String,
+    pub html_url: String,
+    pub followers_url: String,
+    pub following_url: String,
+    pub gists_url: String,
+    pub starred_url: String,
+    pub subscriptions_url: String,
+    pub organizations_url: String,
+    pub repos_url: String,
+    pub events_url: String,
+    pub received_events_url: String,
+    pub site_admin: bool,
+    pub name: Option<String>,
+    pub company: Option<String>,
+    pub blog: Option<String>,
+    pub location: Option<String>,
+    pub email: Option<String>,
+    pub hireable: Option<bool>,
+    pub bio: Option<String>,
+    pub public_repos: Option<u32>,
+    pub public_gists: Option<u32>,
+    pub followers: Option<u32>,
+    pub following: Option<u32>,
+    pub created_at: Option<String>,
+    pub updated_at: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Email {
+    pub email: String,
+    pub primary: bool,
+    pub verified: bool,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Search {
+    pub total_count: u64,
+    pub incomplete_results: bool,
+    pub items: Vec<SearchItem>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct SearchItem {
+    pub name: String,
+    pub path: String,
+    pub sha: String,
+    pub url: String,
+    pub git_url: String,
+    pub html_url: String,
+    pub repository: Repository,
+    pub score: f64,
+}
```

Comment:
I'm amazed there's not a library for a Github client in Rust yet!

Reply:
There are a few but nothing with enough coverage. The good thing is that we are only going to be using a portion of the GitHub API so we can create a well tested sub set and not need to rely on an upstream dependency

Reply:
Admittedly, that _is_ pretty nice. 👍